src/eherkenning/mock/backends.py

```diff
@@ -0,0 +1,67 @@
+import logging
+import re
+
+from django.contrib.auth import get_user_model
+from django.utils.translation import gettext_lazy as _
+
+from digid_eherkenning.backends import BaseBackend
+from digid_eherkenning.utils import get_client_ip
+
+logger = logging.getLogger(__name__)
+
+UserModel = get_user_model()
+
+
+class eHerkenningBackend(BaseBackend):
+    service_name = "eHerkenning"
+    error_messages = dict(
+        BaseBackend.error_messages,
+        **{
+            "eherkenning_no_kvk": _(
+                "Login failed due to no KvK being returned by eHerkenning."
+            ),
+            "eherkenning_len_kvk": _(
+                "Login failed due to no KvK having more then 8 digits."
+            ),
+            "eherkenning_num_kvk": _("Login failed due to no KvK not being numerical."),
+        }
+    )
+
+    def get_or_create_user(self, request, kvk):
+        created = False
+        try:
+            user = UserModel.eherkenning_objects.get_by_kvk(kvk)
+        except UserModel.DoesNotExist:
+            user = UserModel.eherkenning_objects.eherkenning_create(kvk)
+            created = True
+
+        success_message = self.error_messages["login_success"] % {
+            "user": str(user),
+            "user_info": " (new account)" if created else "",
+            "ip": get_client_ip(request),
+            "service": self.service_name,
+        }
+
+        self.log_success(request, success_message)
+
+        return user, created
+
+    def authenticate(self, request, kvk=None):
+        if kvk is None:
+            return
+
+        if kvk == "":
+            self.log_error(request, self.error_messages["eherkenning_no_kvk"])
+            return
+
+        elif not re.match(r"^[0-9]+$", kvk):
```

Comment:
In the last step you continued with `if`, here (and in the next step) you do `elif`, even though there's no difference. I tend to prefer `if` in these situations b/c the implied "else" is not significant.